setup.py

```diff
@@ -288,8 +288,15 @@ def run(self):
     'numpydoc',
 ]
 
+extra_require = {
+    'test:python_version == "2.7"': ['mock'],
```

Comment:
I only found that documented in the[ wheels docs](https://wheel.readthedocs.io/en/latest/#defining-conditional-dependencies)

The idea is that you could run:
`pip install spyder[test]`
and It will install this extra dependencies